buildkite/data_source_team.go

```diff
@@ -0,0 +1,91 @@
+package buildkite
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/shurcooL/graphql"
+)
+
+func dataSourceTeam() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceTeamRead,
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Computed: true,
+				Type:     schema.TypeString,
+			},
+			"slug": {
+				Required: true,
+				Type:     schema.TypeString,
+			},
+			"uuid": {
+				Computed: true,
+				Type:     schema.TypeString,
+			},
+			"name": {
+				Computed: true,
+				Type:     schema.TypeString,
+			},
+			"privacy": {
+				Computed: true,
+				Type:     schema.TypeString,
+			},
+			"description": {
+				Computed: true,
+				Type:     schema.TypeString,
+			},
+			"default_team": {
+				Computed: true,
+				Type:     schema.TypeBool,
+			},
+			"default_member_role": {
+				Computed: true,
+				Type:     schema.TypeString,
+			},
+			"members_can_create_pipelines": {
+				Computed: true,
+				Type:     schema.TypeBool,
+			},
+		},
+	}
+}
+
+// ReadTeam retrieves a Buildkite team
+func dataSourceTeamRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	// Warning or errors can be collected in a slice type
+	var diags diag.Diagnostics
+
+	client := m.(*Client)
+	var query struct {
+		Team TeamNode `graphql:"team(slug: $slug)"`
+	}
```

Comment:
This is a pretty small unit of code, but I wonder if its worth refactoring the team resource so that the reads of both use the same functionality? It would mean the team resource would have to use this logic I think, instead of looking up the Team via GraphQL Relay node ID 🤔

Reply:
Oh yeah very good point on changing the other code.

And we did have plans on adding this and were looking to mimic the AWS provider for lookups on more than just slug. But we can always follow up on that.

I'll try get around to thorough review and testing later this week